engine/class_modules/sc_hunter.cpp

```diff
@@ -983,6 +989,27 @@ struct hunter_action_t: public Base
     return total_regen + floor( total_energize );
   }
 
+  void consume_resource() override
+  {
+    ab::consume_resource();
+
+    if ( !p() -> specs.marksmanship_hunter -> ok() || !p() -> options.t28_4pc )
```

Comment:
Is this the correct way to check that the actor is a MM hunter?

Reply:
We'd typically use `specialization() == HUNTER_MARKSMANSHIP` for that. This should work too though.